datafusion/physical-optimizer/Cargo.toml

```diff
@@ -33,13 +33,17 @@ workspace = true
 
 [dependencies]
 arrow = { workspace = true }
+arrow-array = { workspace = true }
```

Comment:
If possible, can you avoid this explicit dependency? Whatever is used in arrow-array should be available in `arrow` and then we can keep the dependency chain simpler